src/main/java/org/broadinstitute/hellbender/tools/walkers/longreads/CountIndels.java

```diff
@@ -0,0 +1,282 @@
+        private final int           alleleLength;
+        private final String allele;
+
+        public DataEntry(String readName, boolean isNegativeStrand, CigarOperator type, int refPos, int alleleLength, String allele) {
+            this.readName = readName;
+            this.isNegativeStrand = isNegativeStrand;
+            this.type = type;
+            this.refPos = refPos;
+            this.alleleLength = alleleLength;
+            this.allele = allele;
+        }
+
+        public String getReadName() { return readName; }
+        public boolean isNegativeStrand() { return isNegativeStrand; }
+        public CigarOperator getType() { return type; }
+        public int getRefPos() { return refPos; }
+        public int getAlleleLength() { return alleleLength; }
+        public String getAllele() { return allele; }
+
+        @Override
+        public String toString() {
+            return "DataEntry{" +
+                    "readName='" + readName + '\'' +
+                    ", isNegativeStrand=" + isNegativeStrand +
+                    ", type=" + type +
+                    ", refPos=" + refPos +
+                    ", alleleLength=" + alleleLength +
+                    ", allele='" + allele + '\'' +
+                    '}';
+        }
+    }
+
+    private SimpleInterval curInterval = null;
+    private Set<DataEntry> d = null;
+
+    BufferedWriter readStatsWriter, intervalStatsWriter;
+
+    @Override
+    public void onTraversalStart() {
+        try {
+            readStatsWriter = new BufferedWriter(new FileWriter(perReadPathName));
+            readStatsWriter.write(Joiner.on("\t").join(
+                    "interval",
+                    "readName",
+                    "isNegativeStrand",
+                    "lengthOnReference",
+                    "numDeletedBases",
+                    "numInsertedBases",
+                    "delLengths",
+                    "insLengths",
+                    "insAlleles"
+            ) + "\n");
+
+            intervalStatsWriter = new BufferedWriter(new FileWriter(perIntervalPathName));
+            intervalStatsWriter.write(Joiner.on("\t").join(
+                    "interval",
+                    "numReads",
+                    "lengthOnReference",
+                    "allNumDeletedBasesFwd",
+                    "allNumDeletedBasesRev",
+                    "allNumInsertedBasesFwd",
+                    "allNumInsertedBasesRev",
+                    "allDelLengthsFwd",
+                    "allDelLengthsRev",
+                    "allInsAllelesFwd",
+                    "allInsAllelesRev",
+                    "allInsAllelesFwd",
+                    "allInsAllelesRev"
+            ) + "\n");
+        } catch (IOException e) {
+            throw new GATKException(e.getMessage());
+        }
+    }
+
+    @Override
+    public void apply(SimpleInterval interval, ReadsContext readsContext, ReferenceContext referenceContext, FeatureContext featureContext) {
```

Comment:
this reads context has the potential to be absurdly huge and crash if the intervals are too big. This should probably be warned about somewhere.